lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -433,6 +424,206 @@ def add_kp_manifest_data(df: pd.DataFrame, manifest_filenames: tuple, manifest_f
+    """
+    Process clinical data from PHSKC.
+
+    Given a <PHSKC Clinical Data filename> of an Excel document, selects specific
+    columns of interest and reformats the queried data into a stream of JSON
+    documents suitable for the "upload" sibling command.
+
+    All clinical records parsed are output to stdout as newline-delimited JSON
+    records.  You will likely want to redirect stdout to a file.
+    """
+    # specify type of inferred_symptomatic to prevent pandas casting automaticall to boolean
+    clinical_records = pd.read_excel(phskc_filename, dtype={'inferred_symptomatic': 'str'})
+    clinical_records.columns = clinical_records.columns.str.lower()
+
+    clinical_records = trim_whitespace(clinical_records)
+    clinical_records = add_provenance(clinical_records, phskc_filename)
+
+    clinical_records['site'] = 'PHSKC'
+    clinical_records['patient_class'] = 'field'
+    clinical_records['encounter_status'] = 'finished'
+
+    # generate encounter and individual identifiers for each record
+    clinical_records["individual"] = clinical_records.apply(
+        lambda row: generate_patient_hash(
+            row['pat_name'].split(',')[::-1],
+            map_sex(row['sex']),
+            str(row['birth_date']),
+            str(row["pat_address_zip"])
+        ), axis=1
+    )
+
+    clinical_records['identifier'] = clinical_records.apply(
+        lambda row: generate_hash(
+            f"{row['individual']}{row['collect_ts']}".lower()
+        ), axis=1
+    )
+
+    # calculate age based on sample collection date and birth day
+    clinical_records['birth_date'] = pd.to_datetime(clinical_records['birth_date'])
+    clinical_records['age'] = clinical_records.apply(
+        lambda row: age_ceiling(
+                relativedelta(
+                    row['collect_ts'],
+                    row['birth_date']
+                ).years
+            ), axis=1
+    )
+
+    # geocode addresses, then hash them and get the census tract to remove PII from downstream processes
+    with pickled_cache(geocoding_cache_file) as cache:
+        clinical_records['lat'], clinical_records['lang'], clinical_records['canonical_address'] = zip(
+            *clinical_records.apply(
+                lambda row: get_geocoded_address(
+                    {
+                        'street': row['pat_address_line1'],
+                        'secondary': row['pat_address_line2'],
+                        'city': row['pat_address_city'],
+                        'state': row['pat_address_state'],
+                        'zipcode': row['pat_address_zip']
+                    },
+                    cache
+                ),
+                axis=1
+            )
+        )
+
+    db = DatabaseSession()
+    clinical_records = clinical_records.apply(
+       lambda row: encode_addresses(db, row), axis=1
+    )
+
+    column_map = {
+        "collect_ts": "encountered",
```

Comment:
The timestamps in the metadata files _claim_ to be in UTC (they end in "Z" for Zulu), but looking at the values they appear to be in local time.  This is causing them to be shifted by 8 hours when they are ingested.

Reply:
Correction, @bencap pointed out that the values in the metadata files have no timezone info, so are defaulting to UTC. A timezone-aware column is being added to handle it.

Reply:
Great catch on this!